WordPress/gutenberg
# [RNMobile] Social Icons: Ensure inactive icons are visible with block-based themes

* Gutenberg Mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/6302

## What?

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/4780, by ensuring inactive icons within the Social Icons block are always visible within the editor.

## Why?

There are currently certain scenarios where inactive icons in the block are invisible. This happens if a block-based theme with a light background is active, as the native editor displays the light background in contrast with the block's dark mode styles.

## How?

The logic associated with inactive icon styles has been updated so that, instead of using custom colours, we now always use a reduced opacity. This ensures the inactive icons are visible across all types of themes, and also matches the web's behaviour. Further discussion on this approach can be found in the comments here: https://github.com/WordPress/gutenberg/pull/55398#issuecomment-1772693340

## Testing Instructions

* Ensure dark mode is enabled on your device.
* Activate a theme with support for global styles. For example, [Byrne](https://byrnedemo.wordpress.com/).
* While in the site's /wp-admin area in a browser, navigate to the site editor (Appearance → Editor) and select a style with a white background.
* In the app, navigate to the editor for any of the site's posts or pages.
* Verify that the editor's background is white.
* Add a Social Icons block to the editor.
* Verify that the inactive icons in the block are visible.

### Testing Instructions for Keyboard

N/A, solely a visual change.

## Screenshots or screencast 

### Dark mode

| Dark background  | Light background |
| ------------- | ------------- |
| <img src="https://github.com/WordPress/gutenberg/assets/2998162/310a8270-4e18-46d7-ac9f-9d8e0e1463b3" width="100%"> | <img src="https://github.com/WordPress/gutenberg/assets/2998162/5ada628c-4c96-4421-b03f-5fc3dc4ab522" width="100%"> |

### Light mode

| Dark background  | Light background |
| ------------- | ------------- |
| <img src="https://github.com/WordPress/gutenberg/assets/2998162/651b1650-6011-4e87-a723-2dc709d4bbae" width="100%"> | <img src="https://github.com/WordPress/gutenberg/assets/2998162/e39e3dfd-0034-4215-8ea4-a6bfd5620df3" width="100%"> |
